community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/PropertyAndNodeIndexedCheck.java

```diff
@@ -233,4 +195,69 @@ private void checkProperty( NodeRecord record,
             }
         }
     }
+
+    private Object[] getPropertyValues( List<PropertyBlock> properties, int[] indexPropertyIds )
+    {
+        Object[] values = new Object[indexPropertyIds.length];
+        for ( int i = 0; i < indexPropertyIds.length; i++ )
+        {
+            PropertyBlock propertyBlock = propertyWithKey( properties, indexPropertyIds[i] );
+            if ( propertyBlock == null )
+            {
+                throw new IllegalStateException( "We should have checked that the index and node match before coming " +
+                        "here, so this should not happen" );
+            }
+            values[i] = propertyReader.propertyValue( propertyBlock ).value();
+        }
+        return values;
+    }
+
+    private PropertyBlock propertyWithKey( List<PropertyBlock> propertyBlocks, int propertyKey )
+    {
+        for ( PropertyBlock propertyBlock : propertyBlocks )
+        {
+            if ( propertyBlock.getKeyIndexId() == propertyKey )
+            {
+                return propertyBlock;
+            }
+        }
+        return null;
+    }
+
+    private PrimitiveIntSet propertyIds( List<PropertyBlock> propertyBlocks )
+    {
+        PrimitiveIntSet propertyIds = Primitive.intSet();
+        for ( PropertyBlock propertyBlock : propertyBlocks )
+        {
+            propertyIds.add( propertyBlock.getKeyIndexId() );
+        }
+        return propertyIds;
+    }
+
+    private IndexQuery[] seek( LabelSchemaDescriptor schema, Object[] propertyValues )
+    {
+        assert schema.getPropertyIds().length == propertyValues.length;
+        IndexQuery[] query = new IndexQuery[propertyValues.length];
+        for ( int i = 0; i < query.length; i++ )
+        {
+            query[i] = IndexQuery.exact( schema.getPropertyIds()[i], propertyValues[i] );
+        }
+        return query;
+    }
+
+    private PrimitiveLongIterator queryIndexOrEmpty( IndexReader reader, IndexQuery[] query )
+    {
+        PrimitiveLongIterator indexedNodeIds;
+        try
+        {
+            indexedNodeIds = reader.query( query );
+        }
+        catch ( IndexNotApplicableKernelException e )
```

Comment:
Ah, you're right! If we get that exception someone messed up the index implementation, and the only thing to do is likely to crash.